### PR TITLE
Show user type dropdown on age interstitial

### PR DESCRIPTION
**Pull request by ewjordan**
This PR makes it so that the age interstitial now has a user type dropdown, so that it's possible to create oauth accounts that are teachers.

Student in section (user type omitted):
screen shot 2017-07-19 at 10 38 38 am

Teacher selected (shows and requires email field):
screen shot 2017-07-19 at 10 48 55 am

Student selected (shows and requires age field)
screen shot 2017-07-19 at 10 49 01 am

### CODE REVIEW

**Comment by ewjordan** [comment]
@poorvasingal I believe we still need to decide on changing the text before this is ready to go, correct?

File: apps/src/sites/studio/pages/layouts/_age_interstitial.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  $("#email-address").find("input").on('input', function (event) {
+    refreshSubmitState();
+  });
+
+  const sectionsAsStudent = getScriptData('ageInterstitial').hasSections;
+
+  if (sectionsAsStudent) {
+    $("#usertype").hide();
```
Comment (joshlory) [inline apps/src/sites/studio/pages/layouts/_age_interstitial.js:30]: There's enough state change going on here it really feels like this should be a React component.  I know this work is up against a deadline.  Does this need to ship right away, or is there time to build it the "right" way?

**Comment by joshlory** [comment]
The goal is to allow all users to specify their user type at signup, correct?

Can we remove ` || User::TYPE_STUDENT` on this line?  https://github.com/code-dot-org/code-dot-org/blob/85dcd52d69ad4bba696cd71da2de730f317132a3/dashboard/app/models/user.rb#L543

Then the user will fail validation, and they will be taking to then normal signup screen to select an account type.  If account type has already been specified (e.g. by importing from Google Classroom) the user should have the appropriate type already set.

**Comment by ewjordan** [comment]
Closing, going to work from https://github.com/code-dot-org/code-dot-org/pull/16501 instead

**Pull request closed without merging**